### PR TITLE
Use autoloading for PHPCR-API tests

### PR DESCRIPTION
**Pull request by dantleech**
See: https://github.com/phpcr/phpcr-api-tests/pull/161

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     },
     "require-dev": {
         "psr/log": "~1.0",
-        "phpcr/phpcr-api-tests": "2.1.4",
+        "phpcr/phpcr-api-tests": "dev-clean@dev",
```
Comment (dantleech) [inline composer.json]: Remember to revert this.

**Comment by lsmith77** [comment]
have you also tested this with Jackrabbit?

**Comment by dantleech** [comment]
Have just run the tests now and they're fine.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -13,10 +13,6 @@
 $loader->add('Jackalope\\Test', __DIR__);
 $loader->add('Jackalope', __DIR__ . '/../vendor/jackalope/jackalope/tests');
```
Comment (dbu) [inline tests/bootstrap.php]: do we still need these lines? should those not be autoloaded as well?

File: composer.json (hunk 1 above)
Comment (dbu) [inline composer.json]: change to 2.1.5 and jackalope/jackalope to 1.2.1 once https://github.com/phpcr/phpcr-api-tests/pull/163 is merged and we tagged a new version

File: tests/bootstrap.php (hunk 2 above)
Comment (dbu) [inline tests/bootstrap.php]: ah, the first one yet, but this one not because the jackalope tests should not be autoloaded for consumers of the library

Comment (dantleech) [inline tests/bootstrap.php]: hmm.. but thats what `autoload-dev` is for in composer?

Comment (dbu) [inline tests/bootstrap.php]: only for the library itself afaik. i think jackalope-doctrine-dbal does
not use the autoload-dev from its dependencies like jackalope/jackalope

**Pull request merged**